linkedlist.ts

```diff
@@ -34,8 +34,42 @@ export default class LinkedList<V> {
     return value;
   }
 
+  filter(filter: (item: V) => boolean) {
+    let last: LinkedNode<V> | null;
+    let next: LinkedNode<V> | null;
+    while (this._head !== null && !filter(this._head.value)) {
+      this._head = this._head.next
+      this._length--;
+    }
+    if (this._head === null) {
+      this._tail = null;
+      return
+    }
+    last = this._head;
+    next = this._head.next;
+    while (next !== null) {
+      if (filter(next.value)) {
+        last = next;
+        next = next.next
+      } else {
+        next = next.next
+        last.next = next
+        this._length--;
+      }
+    }
+    this._tail = last;
+  }
+
   clear() {
     this._length = 0;
     this._head = this._tail = null;
   }
+
+  *[Symbol.iterator]() {
```

Comment:
We should clarify that this iterator does not keep yielding items as they are pushed into the list. It synchronously runs until the current end of the list and that's it.

I'm only pointing this out because I'm working on a cursor-like object that will never actually end, continuously moving to new items as they are added to the list, as the backing for the revised version of `HistoryReader`. We should be very explicit about the difference between these two.